plugins/experimental/buffer_upload/buffer_upload.cc

```diff
@@ -749,10 +753,22 @@ attach_pvc_plugin(TSCont /* contp ATS_UNUSED */, TSEvent event, void *edata)
         }
 
         char replacement_host_str[host_hdr_str_val_len + 1];
-        memset(replacement_host_str, 0, sizeof replacement_host_str);
         memcpy(replacement_host_str, host_hdr_str_val, host_hdr_str_val_len);
+        replacement_host_str[host_hdr_str_val_len] = '\0';
         TSDebug(DEBUG_TAG, "Adding host to request url: %s", replacement_host_str);
 
+        const char *colon = strchr(replacement_host_str, ':');
+        if (colon != NULL && colon + 1 != NULL) {
```

Comment:
I think you mean `colon[1] != 0` (or, in Bryan's lingo, `'\0\'`). Because if `colon` is not `NULL` then `colon+1` is not going to be either.